src/main/java/com/sequenceiq/cloudbreak/service/stack/handler/StackUpdateSuccessHandler.java

```diff
@@ -45,7 +45,8 @@ public void accept(Event<StackUpdateSuccess> t) {
         LOGGER.info("Accepted {} event.", ReactorConfig.STACK_UPDATE_SUCCESS_EVENT);
         Set<String> instanceIds = updateSuccess.getInstanceIds();
         if (updateSuccess.isRemoveInstances()) {
-            stackUpdater.updateNodeCount(stackId, stack.getNodeCount() - instanceIds.size());
+            stackUpdater.updateNodeCount(stackId, stack.getTemplateAsGroup(updateSuccess.getHostGroup()).getNodeCount()  - instanceIds.size(),
+                    updateSuccess.getHostGroup());
```

Comment:
We are updating an InstanceGroup domain object here, why is it needed to update it through the Stack updater and resave the whole Stack domain?